src/MASclient.ts

```diff
@@ -0,0 +1,158 @@
+limitations under the License.
+*/
+
+import { ClientCredentials, AccessToken } from "simple-oauth2";
+import { IConfig } from "./config";
+import axios from "axios";
+import { LogService } from "@vector-im/matrix-bot-sdk";
+
+export class MASclient {
+    public readonly config: IConfig;
+    private client: ClientCredentials;
+    private accessToken: AccessToken;
+
+    constructor(config: IConfig) {
+        LogService.info("MAS client", "Setting up mas client");
+        this.config = config;
+        const clientConfig = {
+            client: {
+                id: config.mas.clientId,
+                secret: config.mas.clientSecret,
+            },
+            auth: {
+                tokenPath: config.mas.url + "/oauth2/token",
+                tokenHost: config.mas.url,
+            },
+        };
+        this.client = new ClientCredentials(clientConfig);
+    }
+
+    public async getAccessToken() {
+        if (!this.accessToken || this.accessToken.expired()) {
+            // fetch a new one
+            const tokenParams = { scope: "urn:mas:admin" };
+            try {
+                this.accessToken = await this.client.getToken(tokenParams);
+            } catch (error) {
+                LogService.error("MAS client", "Error fetching auth token for MAS:", error.message);
+                throw error;
+            }
+        }
+        return this.accessToken;
+    }
+
+    public async getMASUserId(userId: string): Promise<string> {
+        const index = userId.indexOf(":");
+        const localpart = userId.substring(1, index);
+        const accessToken = await this.getAccessToken();
+        try {
+            const resp = await axios({
+                method: "get",
+                url: this.config.mas.url + `/api/admin/v1/users/by-username/${localpart}`,
+                headers: {
+                    "User-Agent": "Mjolnir",
+                    "Content-Type": "application/json; charset=UTF-8",
+                    "Authorization": `Bearer ${accessToken.token.access_token}`,
+                },
+            });
+            return resp.data.data.id;
+        } catch (error) {
+            LogService.error("MAS client", `Error fetching MAS id for user ${userId}:`, error.message);
+            throw error;
+        }
+    }
+
+    public async deactivateMasUser(userId: string): Promise<void> {
+        const masId = await this.getMASUserId(userId);
+        const accessToken = await this.getAccessToken();
+        const url = this.config.mas.url + `/api/admin/v1/users/${masId}/deactivate`;
+        try {
+            await axios({
+                method: "post",
+                url: url,
+                headers: {
+                    "User-Agent": "Mjolnir",
+                    "Content-Type": "application/json; charset=UTF-8",
+                    "Authorization": `Bearer ${accessToken.token.access_token}`,
+                },
+            });
```

Comment:
moving the axios calls to a `doRequest` function would be best, so if we need to update something it's a single code site. The `getAccessToken()` call can be in that function too.